mitodl/open-discussions
# Display cover image thumbnail on compact post display

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1595

#### What's this PR do?
- Modifies `CompactPostDisplay` component to display the cover image thumbnail for an article if present.
- On the backend, crops the image before creating the article cover thumbnail to mimic embedly's API for external link-post thumbnails.

#### How should this be manually tested?
- Add an article with cover image via django shell:
```python
import os
import json
from django.contrib.auth.models import User
from django.urls import reverse
from rest_framework.test import APIClient
from channels.models import Channel, Post, Article

client = APIClient()
client.force_login(User.objects.get(username='<username>'))
channel = Channel.objects.get(name="<channel_name>")
article_content = [{"key": "value", "nested": {"number": 4}}]
img_file = "static/images/avatar_default.png"
url = reverse("post-list", kwargs={"channel_name": channel.name})
with open(img_file, "rb") as f:
    resp = client.post(
        url, 
        {"title": "foo", "article_content": json.dumps(article_content), "cover_image": f}, 
        format="multipart"
    )

post = Post.objects.get(post_id=resp.json()["id"])
print(post.article.cover_image_small.url)
 ```
- Go to the article's channel page.  The thumbnail should be visible and have the same dimensions as a link post thumbnail.

#### Screenshots (if appropriate)
Two link-posts and two articles were created using the same images for thumbnails.  

**Desktop channel page**
<img width="661" alt="screen shot 2018-12-27 at 11 21 12 am" src="https://user-images.githubusercontent.com/187676/50487500-a784fe00-09cc-11e9-8b64-fcb80837703f.png">

**Desktop frontpage - wide**
<img width="975" alt="screen shot 2018-12-27 at 12 05 32 pm" src="https://user-images.githubusercontent.com/187676/50488311-dea8de80-09cf-11e9-9028-9b51ae90771d.png">


**Mobile:**
<img width="374" alt="screen shot 2018-12-27 at 11 21 42 am" src="https://user-images.githubusercontent.com/187676/50487499-a784fe00-09cc-11e9-8b5e-688552fbf4dc.png">
